docs/dev/framework/security.md

```diff
@@ -0,0 +1,275 @@
+[access control][SymfonyAccessControl] via the `contao/manager-bundle`. If you do not use the Managed Edition and added Contao to your
+custom Symfony application, you will have to register Contao's security settings yourself, as mentioned in the 
+[Getting Started article][ContaoConfiguration]. Contao's Symfony Security configuration [in Contao **4.9**][Contao49SecurityYaml] looks like this for example:
+
+{{% expand "Contao's Symfony Security Configuration" %}}
+```yaml
+security:
+    providers:
+        contao.security.backend_user_provider:
+            id: contao.security.backend_user_provider
+
+        contao.security.frontend_user_provider:
+            id: contao.security.frontend_user_provider
+
+    encoders:
+        Contao\User:
+            algorithm: auto
+
+    firewalls:
+        dev:
+            pattern: ^/(_(profiler|wdt|error)|css|images|js)/
+            security: false
+
+        contao_install:
+            pattern: ^/contao/install$
+            security: false
+
+        contao_backend:
+            request_matcher: contao.routing.backend_matcher
+            provider: contao.security.backend_user_provider
+            user_checker: contao.security.user_checker
+            anonymous: ~
+            switch_user: true
+
+            contao_login:
+                remember_me: false
+
+            logout:
+                path: contao_backend_logout
+                handlers:
+                    - contao.security.logout_handler
+                    - contao_manager.security.logout_handler
+                success_handler: contao.security.logout_success_handler
+
+        contao_frontend:
+            request_matcher: contao.routing.frontend_matcher
+            provider: contao.security.frontend_user_provider
+            user_checker: contao.security.user_checker
+            anonymous: ~
+            switch_user: false
+
+            contao_login:
+                remember_me: true
+
+            remember_me:
+                secret: '%kernel.secret%'
+                remember_me_parameter: autologin
+
+            logout:
+                path: contao_frontend_logout
+                target: contao_root
+                handlers:
+                    - contao.security.logout_handler
+                success_handler: contao.security.logout_success_handler
+
+    access_control:
+        - { path: ^/contao/login$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/contao/logout$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/contao(/|$), roles: ROLE_USER }
+        - { path: ^/, roles: [IS_AUTHENTICATED_ANONYMOUSLY] }
+```
+{{% /expand %}}
+
+If you want to learn more about Symfony's Security Component use the provided links to read up on. This documentation will only cover
+implmentation details that are unique to Contao.
+
+Since within Contao you can put a login form on basically any page, Contao does not utilise Symfony's built-in 
```

Comment:
`form_login`, `request matcher`, `user provider` and `user checker` are not related to each other at all.

1. Request matchers allow us to not match on paths but on the scope attribute (anything because request is matched through a service). The idea behind this was also to allow any other custom paths in your Symfony app that won't be affected by the Contao firewall.

2. form_login is the default Symfony username/password factory, but it does not work for Contao because we don't have known login paths (specifically in the front end). We match on `FORM_SUBMIT = tl_login` instead, and since we have our own firewall factory we do a bunch of other stuff as well (mostly hardcoding stuff instead of having a firewall configuration).

3. user provider is something you always need with a firewall, so not really Contao-specific

4. user checker is Symfony-default as well. it _could_ be optional but having one is pretty standard, you can e.g. check if a user has been disabled to throw him out of an active login session.

Reply:
Ad 1) & 2): that's what I want to say with that sentence. Regarding 3) and 4) I'll flesh this out more.